src/PlaneWaveBasis.jl

```diff
@@ -244,6 +277,14 @@ function PlaneWaveBasis(model::Model, Ecut::Number;
     PlaneWaveBasis(model, Ecut, kcoords, ksymops, symmetries; kwargs...)
 end
 
+"""
+Sum an array over kpoints, taking weights into account
+"""
+function weighted_ksum(basis, arr)
+    res = sum(basis.kweights .* arr)
+    mpi_sum(res, basis.comm_k)
+end
```

Comment:
Maybe also add a function `weighted_ksum(f::Function, arr, basis)` which first applies `f` to `arr`?

Reply:
We don't really use it?

Reply:
Sure, but only since you have not thought about it. Actually it could be used in *all* instances where you employ `weighted_ksum` instead of the constructs you have right now (and it could even avoid allocations by passing it right through to `sum`).

Reply:
Well if we use it we add it. Otherwise it's dead code

Reply:
Well in some sense that's my point: We can use it sensibly, but fair enough. I might give it a try and see what it looks like, though.

Reply:
I agree we could probably replace some of the reduce code we have (eg in the energy computations) with a do block using this. It avoids explicit mpi calls.